app/models/user/permissions.rb

```diff
@@ -5,7 +5,9 @@
 module User::Permissions
   extend ActiveSupport::Concern
 
-  ATTRIBUTES = %I[ role member_permission_ids member_permission_service_ids ]
+  ATTRIBUTES = %I[role member_permission_ids member_permission_service_ids].freeze
+  # The value that can be used for setting the member permission service ids to empty array
+  EMPTY_VALUE = "[]"
```

Comment:
Yeah, me neither...
I tried to explain here: https://github.com/3scale/porta/pull/3929#discussion_r1835321872

Indeed, it is for backwards compatibility, but in fact any set of characters that is not an actual integer number works the same way as this "special" character, so maybe we can treat them all the same way. WDYT?

Reply:
If you send any non-numerical character it will be converted to `0`, so for me the easiest way would be to just remove zeroes from the resulting `service_ids` array, rather than explicitly considering an special character. Then in the docs you can tell users to send `"[]"`, that's fine, but no need to be considered by backend at all. In fact, docs mentioning `"[]"` are not accurate because the endpoint accepts `"[]"` and any non-numerical character.

So I think it was fine before the PR, but you want `0` to be a valid service number, then it's more complicated, because receiving a non-integer character is now interpreted as allowing permissions over service `0`. We could maybe use use `Integer()` for this:

```
Loading development environment (Rails 6.1.7.9)
[1] pry(main)> Integer("   5   ")
=> 5
[2] pry(main)> Integer("0")
=> 0
[3] pry(main)> Integer("[]")
ArgumentError: invalid value for Integer(): "[]"
from (pry):3:in `Integer'
[4] pry(main)> Integer("24gt")
ArgumentError: invalid value for Integer(): "24gt"
from (pry):4:in `Integer'
[5] pry(main)> Integer("")
ArgumentError: invalid value for Integer(): ""
from (pry):5:in `Integer'
[6] pry(main)> 

```

`map` the given service ids, pass them through `Integer()`, return `nil` on exceptions, then compact the result. That would reject al non-integer characters but accept `0`.

Summarizing, I would:

- State in the docs that `"[]"` means "no services" and any non-numerical character will be removed
- Don't care about `"[]"` at all in the backend
- Use `Integer()` to process the given values

Reply:
Thanks a lot for the suggestions @jlledom!

> `map` the given service ids, pass them through `Integer()`, return `nil` on exceptions, then compact the result. That would reject al non-integer characters but accept `0`.

I agree, and I've done it here: https://github.com/3scale/porta/pull/3929/commits/06db7c69148f6aaac57921b0a6127f3704e838c4


> State in the docs that `"[]"` means "no services" and any non-numerical character will be removed

I don't actually want to document `[]` as a special value. I think it is better to say to pass an empty value instead.